crates/compiler/load_internal/src/file.rs

```diff
@@ -2421,17 +2336,55 @@ fn update<'a>(
                 }
             }
 
-            check_for_missing_package_shorthand_in_cache(&header, &state.arc_shorthands)?;
+            work.insert((home, Phase::Parse));
+
+            state.module_cache.headers.insert(header.module_id, header);
+
+            start_tasks(arena, &mut state, work, injector, worker_listeners)?;
+
+            let work = state.dependencies.notify(home, Phase::LoadHeader);
+
+            start_tasks(arena, &mut state, work, injector, worker_listeners)?;
+
+            Ok(state)
+        }
+        Parsed(parsed) => {
+            let module_id = parsed.module_id;
+
+            if let HeaderType::Platform {
+                config_shorthand, ..
+            } = parsed.header_type
+            {
+                let undefined_shorthands: Vec<_> = parsed
+                    .package_qualified_imported_modules
+                    .iter()
+                    .filter(|pqim| match pqim {
+                        PackageQualified::Unqualified(_) => false,
+                        PackageQualified::Qualified(shorthand, _) => {
+                            !(parsed.packages.contains_key(shorthand)
+                                || shorthand == &config_shorthand)
+                        }
+                    })
+                    .collect();
+
+                // shorthands must be defined by the module!
+                assert!(
+                    undefined_shorthands.is_empty(),
+                    "{undefined_shorthands:?} not in {:?} ",
+                    &parsed.packages
+                );
+            }
```

Comment:
this is just an expected-state assertion right? If so consider lifting it out to a function instead of inlining here.

Reply:
Actually, this should be validated already, we don't need this anymore.